### PR TITLE
thanos: upgrade dependency and default to v0.5.0

### PR DESCRIPTION
**Pull request by abursavich**
Per https://github.com/coreos/prometheus-operator/pull/2623#issuecomment-499153323 and https://github.com/coreos/prometheus-operator/pull/2629#issuecomment-499270790, this upgrades the thanos dependency and default to v0.5.0.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,22 +14,19 @@ require (
...
+	github.com/improbable-eng/thanos v0.5.0
 	github.com/jsonnet-bundler/jsonnet-bundler v0.1.0
 	github.com/kylelemons/godebug v0.0.0-20170820004349-d65d576e9348
 	github.com/mitchellh/hashstructure v0.0.0-20170609045927-2bca23e0e452
 	github.com/oklog/run v1.0.0
 	github.com/pkg/errors v0.8.1
-	github.com/prometheus/client_golang v0.9.2
-	github.com/prometheus/prometheus v0.0.0-20190328180107-4d60eb36dcbe
+	github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
+	github.com/prometheus/prometheus v2.9.2+incompatible
```
Comment (abursavich) [inline go.mod]: I hadn't seen the `+incompatible` suffix before and it worried me for a minute, but it's okay: https://github.com/golang/go/wiki/Modules#can-a-module-consume-a-v2-package-that-has-not-opted-into-modules-what-does-incompatible-mean

**Comment by abursavich** [comment]
cc @brancz

**Comment by brancz** [comment]
Thank you so much for your awesome contributions! lgtm on green :+1:

**Comment by abursavich** [comment]
I have no idea why that test would fail that way... If it's a real issue and not just a flaky test, it may be a few days before I can get to it.

**Comment by brancz** [comment]
That was a known flake. Restarted.

**Comment by brancz** [comment]
Thanks lgtm :+1:

**Pull request merged**